common/perf-test-core/src/main/java/com/azure/perf/test/core/ApiPerfTestBase.java

```diff
@@ -225,6 +229,72 @@ public Mono<Void> runAllAsync(long endNanoTime) {
             .then();
     }
 
+    @Override
+    public CompletableFuture<Void> runAllAsyncWithCompletableFuture(long endNanoTime) {
+        completedOperations = 0;
+        lastCompletionNanoTime = 0;
+        long startNanoTime = System.nanoTime();
+        Semaphore semaphore = new Semaphore(10); // Limit to 10 concurrent tasks
```

Comment:
Why do we have a hard coded limit of 10? Should this be configurable at the perf test level?

Reply:
updated it to be used with configurable options.